test/test_geo_interface.jl

```diff
@@ -207,71 +209,107 @@ using Plots
-        geom = convert(MultiLineString, XMultiLineString())
+        geom = GeoInterface.convert(MultiLineString, GeoInterface.MultiLineString(coords))
         @test geom isa MultiLineString
         @test GeoInterface.coordinates(geom) == coords
+        for f in one_arg_functions
+            @test f(LibGEOS.MultiLineString(coords)) == f(GeoInterface.MultiLineString(coords))
+        end
+        coords2 = [[[0.0, 10], [0.5, 10], [20.0, 20], [10.0, 10], [0.0, 10]]]
+        for f in two_arg_functions
+            @test f(LibGEOS.MultiLineString(coords), LibGEOS.MultiLineString(coords2)) ==
+                  f(GeoInterface.MultiLineString(coords), LibGEOS.MultiLineString(coords2))
+        end
 
-        struct XPolygon end
         coords = [[[0.0, 0], [0.0, 10], [10.0, 10], [10.0, 0], [0.0, 0]]]
-        GeoInterface.geomtrait(::XPolygon) = GeoInterface.PolygonTrait()
-        GeoInterface.coordinates(::XPolygon) = coords
-        geom = convert(Polygon, XPolygon())
+        geom = GeoInterface.convert(Polygon, Polygon(coords))
         @test geom isa Polygon
         @test GeoInterface.ngeom(geom) == 1
         @test GeoInterface.nring(geom) == 1
         @test GeoInterface.nhole(geom) == 0
         @test GeoInterface.coordinates(geom) == coords
+        for f in one_arg_functions
+            @test f(LibGEOS.Polygon(coords)) == f(GeoInterface.Polygon(coords))
+        end
+        coords2 = [[[0.0, 10], [0.5, 10], [20.0, 20], [10.0, 10], [0.0, 10]]]
+        for f in two_arg_functions
+            @test f(LibGEOS.Polygon(coords), LibGEOS.Polygon(coords2)) ==
+            f(GeoInterface.Polygon(coords), LibGEOS.Polygon(coords2))
+        end
 
         pgeom = LibGEOS.prepareGeom(geom)
         @test GeoInterface.coordinates(pgeom) == coords
         LibGEOS.destroyGeom(pgeom)
 
-        struct XMultiPolygon end
         coords = [[[[0.0, 0], [0.0, 10], [10.0, 10], [10.0, 0], [0.0, 0]]]]
-        GeoInterface.geomtrait(::XMultiPolygon) = GeoInterface.MultiPolygonTrait()
-        GeoInterface.coordinates(::XMultiPolygon) = coords
-        geom = convert(MultiPolygon, XMultiPolygon())
+        geom = GeoInterface.convert(MultiPolygon, GeoInterface.MultiPolygon(coords))
         @test geom isa MultiPolygon
         @test GeoInterface.coordinates(geom) == coords
+        for f in one_arg_functions
+            @test f(LibGEOS.MultiPolygon(coords)) == f(GeoInterface.MultiPolygon(coords))
+        end
+        coords2 = [[[[0.0, 10], [0.5, 10], [20.0, 20], [10.0, 10], [0.0, 10]]]]
+        for f in two_arg_functions
+            @test f(LibGEOS.MultiPolygon(coords), LibGEOS.MultiPolygon(coords2)) ==
+                  f(GeoInterface.MultiPolygon(coords), LibGEOS.MultiPolygon(coords2))
+        end
 
         struct XMesh end
         GeoInterface.geomtrait(::XMesh) = GeoInterface.PolyhedralSurfaceTrait()
-        @test_throws Exception convert(MultiPolygon, XMesh())
-
+        @test_throws Exception GeoInterface.convert(MultiPolygon, XMesh())
```

Comment:
Shall we remove the last of the `struct X` as well? This test doesn't seem very useful.